client/layout/tracks-opt-out/index.jsx

```diff
@@ -0,0 +1,53 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import debug from 'debug';
+import PropTypes from 'prop-types';
+import React from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { setTracksOptOut } from 'state/analytics/actions';
+
+/**
+ * Module variables
+ */
+const tracksDebug = debug( 'calypso:analytics:tracks' );
+
+class TracksOptOut extends React.Component {
+	static propTypes = {
+		userSettings: PropTypes.object.isRequired,
+		setTracksOptOut: PropTypes.func.isRequired,
+	};
+
+	componentDidMount() {
+		this.props.userSettings.on( 'change', this.syncTracksOptOut );
+	}
+
+	componentWillUnmount() {
+		this.props.userSettings.off( 'change', this.syncTracksOptOut );
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		if ( this.props.userSettings !== nextProps.userSettings ) {
```

Comment:
We should probably check if the opt out bit specifically has changed. This is a ref check so if we return a new userSettings object with no changes this will still trigger.

Reply:
We need to keep this in case the `userSettings` is updated and we want to re-attach ourselves to its changes.

Reply:
To clarify, the updated check would look something like:
```jsx
if ( this.props.userSettings !== nextProps.userSettings ) {
    const optOut = this.props.userSettings.getSetting( 'tracks_opt_out' );
    const nextOptOut = nextProps.userSettings.getSetting( 'tracks_opt_out' );
    if ( optOut === nextOptOut ) {
        //... early return
        return;
    }
    // call sync otherwise
```

Reply:
`userSettings` is a global singleton (the default export of `lib/user-settings`) that is always the same object. There is no need to detect changes and reattach listeners -- the change will never happen.